packages/common/src/downloads/fetchDownloadFileUrl.ts

```diff
@@ -0,0 +1,32 @@
+import HubError from "../HubError";
+import { canUseExportImageFlow } from "./helpers/canUseExportImageFlow";
+import { canUseExportItemFlow } from "./helpers/canUseExportItemFlow";
+import { canUseHubDownloadApi } from "./helpers/canUseHubDownloadApi";
+import { fetchExportImageDownloadFileUrl } from "./helpers/file-url-fetchers/fetchExportImageDownloadFileUrl";
+import { fetchExportItemDownloadFileUrl } from "./helpers/file-url-fetchers/fetchExportItemDownloadFileUrl";
+import { fetchHubApiDownloadFileUrl } from "./helpers/file-url-fetchers/fetchHubApiDownloadFileUrl";
+import { IFetchDownloadFileUrlOptions } from "./helpers/types";
+
+/**
+ * Fetches a download file URL for the given entity and format.
+ * @param options options to refine / filter the results of the fetchDownloadFileUrl operation
+ * @returns a promise that resolves with the download file URL
+ */
+export async function fetchDownloadFileUrl(
+  options: IFetchDownloadFileUrlOptions
+): Promise<string> {
+  let fetchingFn;
+  if (canUseHubDownloadApi(options.entity, options.context)) {
+    fetchingFn = fetchHubApiDownloadFileUrl;
+  } else if (canUseExportItemFlow(options.entity)) {
+    fetchingFn = fetchExportItemDownloadFileUrl;
+  } else if (canUseExportImageFlow(options.entity)) {
+    fetchingFn = fetchExportImageDownloadFileUrl;
```

Comment:
consider using dynamic `import()` for these, for example:

`fetchingFn = (await import('./helpers/file-url-fetchers/fetchHubApiDownloadFileUrl')).fetchHubApiDownloadFileUrl`